src/test/java/com/rocketden/main/service/RoomServiceTests.java

```diff
@@ -0,0 +1,102 @@
+package com.rocketden.main.service;
+
+import com.rocketden.main.dao.RoomRepository;
+import com.rocketden.main.dto.room.CreateRoomRequest;
+import com.rocketden.main.dto.room.CreateRoomResponse;
+import com.rocketden.main.dto.room.JoinRoomRequest;
+import com.rocketden.main.dto.room.JoinRoomResponse;
+import com.rocketden.main.model.Room;
+import org.junit.Before;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class RoomServiceTests {
+
+    @Mock
+    private RoomRepository repository;
+
+    @Spy
+    @InjectMocks
```

Comment:
Two things happening here:
1. Mock and inject the RoomRepository field variable into RoomService so no database is actually touched
2. Create a spy of the RoomService allowing you to specify a set return value when `generateRoomId` is called

More on spies vs. mocks here: https://www.baeldung.com/mockito-spy#mock-vs-spy-in-mockito

Reply:
Thanks so much for the explanation, very helpful!